### PR TITLE
Prevents users from logging out before accepting invitations to join a team

### PR DESCRIPTION
**Pull request by daniellemaxwell**
### Scope of changes

Prevents users from having to log out to accept an invitation request while logged into Beacon.

Fixes SC-21664

Note: When the user accepts an invite, a new tab or window opens, and they're logged out of their previous session. I can add a follow on story that displays an error explaining why the user is logged out if they return to the previous tab or window.

### Type of change

- [x] new feature
- [ ] bug fix
- [ ] documentation
- [ ] testing
- [ ] technical debt
- [ ] other (describe)

### Acceptance criteria

https://www.awesomescreenshot.com/video/22679550?key=d116cb41d88cda4cfa85bfdc929ad5de

### Definition of Done

- [x] I have manually tested the change running it locally (having rebuilt all containers) or via unit tests 
- [ ] I have added unit and/or integration tests that cover my changes
- [ ] I have added new test fixtures as needed to support added tests
- [ ] I have updated the dependencies list if necessary (including updating yarn.lock and/or go.sum)
- [ ] I have recompiled and included new protocol buffers to reflect changes I made if necessary
- [x] Check this box if a reviewer can merge this pull request after approval (leave it unchecked if you want to do it yourself)
- [x] I have notified the reviewer via Shortcut or Slack that this is ready for review
- [ ] Front-end: Checked sm, md, lg screen resolutions for effective responsiveness
- [ ] Backend-end: Documented service configuration changes or created related devops stories

### Reviewer(s) checklist

- [ ] Front-end: I've reviewed the Figma design and confirmed that changes match the spec.
- [ ] Any new user-facing content that has been added for this PR has been QA'ed to ensure correct grammar, spelling, and understandability.
- [ ] Are there any TODOs in this PR that should be turned into stories?

### CODE REVIEW

File: web/beacon-app/src/features/teams/components/ExistingUserInvitationPage.tsx [hunk 1]
```diff
@@ -25,30 +28,47 @@ export default function ExistingUserInvitationPage({ data }: { data: any }) {
...
+      const token = decodeToken(login.auth.access_token) as any;
+      Store.setAuthUser(token, !!login.authenticated);
+
+      navigate(APP_ROUTE.DASHBOARD);
+    }
   }
+
   return (
     <div>
-      {login.hasAuthFailed && (
```
Comment (daniellemaxwell) [inline web/beacon-app/src/features/teams/components/ExistingUserInvitationPage.tsx:45]: This change removed a duplicate toast message.

File: web/beacon-app/src/features/teams/api/invitationAuthenticationRequest.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+): ApiAdapters['getInvitationAuthentication'] {
+  return async (token: string) => {
+    const response = (await request(`${APP_ROUTE.INVITE}/accept`, {
+      method: 'POST',
+      data: {
+        token,
+      },
+    })) as any;
+
+    return getValidApiResponse<UserAuthResponse>(response);
```
Comment (daniellemaxwell) [inline web/beacon-app/src/features/teams/api/invitationAuthenticationRequest.ts:17]: The `UserAuthResponse` is listed [here](https://github.com/rotationalio/ensign/blob/develop/web/beacon-app/src/features/auth/types/LoginService.ts).

**Review by pdeziel — APPROVED** [review]
Thanks for fixing this - I think logging the user out of the previous tab is a reasonable side effect for now since that session would be expired anyway. It actually might be a good idea to add a generic "your session has expired, please login again" message to the login page, I'm not sure if we have that yet but that's what other sites do.

I attempted to locally reproduce the 500 error you were seeing but I had a problem where the verify email page was blank. I'm not sure if that's only on my end so it might be a good idea to try that again.

I did notice that `/verify` was being called twice as well...so there might be a larger issue we've discovered with the frontend issuing duplicate requests to the backend? If that's the case I'm satisfied with opening a bug story to look into that from the frontend perspective so we can get this merged in.

**Comment by daniellemaxwell** [comment]
@pdeziel Were you directed to the `/verify` page instead of the `/invite` page after accepting an invite that was sent to an existing user? That shouldn't happen, so I want to be sure that I can reproduce the error.

**Pull request merged**